packages/components/page-header/src/page-header.tsx

```diff
@@ -26,85 +21,107 @@ const {
   pageHeaderBackIconWrapperClsx,
 } = pageHeader
 
-const PageHeader = (props: PageHeaderProps) => {
-  const {
-    title,
-    titleClassName,
-    onBack,
-    titleExtra,
-    titleExtraClassName,
-    actions = [],
-    actionsWrapperClassName,
-    className,
-    ...rest
-  } = props
-
-  const backButton = (
+const PageHeaderBackButton = ({
+  onBack,
```

Comment:
Don't expose `onBack`, instead, expose all necessary props of the IconButton (make sure we have `as` prop so we can use `Link` instead of push method.